### PR TITLE
[doc][hackathon] To add Adadelta Optimizer to the documentation

### PR DESCRIPTION
**Pull request by iramazanli**
It has been discussed before that adding description of Optimization algorithms to PyTorch Core documentation may result in a nice Optimization research tutorial. In the following tracking issue we mentioned about all the necessary algorithms and links to the originally published paper  https://github.com/pytorch/pytorch/issues/63236.

In this PR we are adding description of AdaDelta Algorithm to the documentation.  For more details, we refer to the paper  here https://arxiv.org/abs/1212.5701

AdaDeltaalg

cc @vincentqb @iramazanli

### CODE REVIEW

File: torch/optim/adadelta.py [hunk 1]
```diff
@@ -5,9 +5,32 @@
...
-
-    It has been proposed in `ADADELTA: An Adaptive Learning Rate Method`__.
+    r"""Implements Adadelta algorithm.
+
+    .. math::
+       \begin{aligned}
+            &\rule{110mm}{0.4pt}                                                                 \\
+            &\textbf{input}      : \gamma \text{ (lr)}, \: \theta_0 \text{ (params)},
+                \: f(\theta) \text{ (objective)}, \: \rho \text{ (decay)}, \: weightdecay        \\
+            &\textbf{initialize} :  E[g^2]_0  \leftarrow 0, \: E[\Delta x^2]_0  \leftarrow 0
```
Comment (albanD) [inline torch/optim/adadelta.py:15]: Why not name these like the other optimizers (like exp_avg) I feel like this would be easier to read?

Comment (iramazanli) [reply torch/optim/adadelta.py]: sounds good ! and done

File: torch/optim/adadelta.py [hunk 2]
```diff
@@ -5,9 +5,33 @@
...
-    It has been proposed in `ADADELTA: An Adaptive Learning Rate Method`__.
+    r"""Implements Adadelta algorithm.
+
+    .. math::
+       \begin{aligned}
+            &\rule{110mm}{0.4pt}                                                                 \\
+            &\textbf{input}      : \gamma \text{ (lr)}, \: \theta_0 \text{ (params)},
+                \: f(\theta) \text{ (objective)}, \: \rho \text{ (decay)},
+                \: \lambda \text{ (weight decay)}                                                \\
+            &\textbf{initialize} : square\_avg_0 \leftarrow 0,
```
Comment (albanD) [inline torch/optim/adadelta.py:16]: Why not use `v_0` like the other algorithms: https://pytorch.org/docs/master/generated/torch.optim.RMSprop.html#torch.optim.RMSprop ?

Comment (iramazanli) [reply torch/optim/adadelta.py]: done !

**Review by albanD — APPROVED** [review]

**Pull request closed without merging**